File: L2ScriptMaker/HtmlButtonDesigner.vb
Public Class HtmlButtonDesigner

    Private Sub QuitButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles QuitButton.Click
        Me.Close()
    End Sub

    Private Sub StartButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles StartButton.Click

        ' check input params
        If WidthValue.Text = "" Then MessageBox.Show("No Width param", "No param", MessageBoxButtons.OK, MessageBoxIcon.Error) : Exit Sub
        If HeightValue.Text = "" Then MessageBox.Show("No Height param", "No param", MessageBoxButtons.OK, MessageBoxIcon.Error) : Exit Sub
        If BackValue.Text = "" Then MessageBox.Show("No Back param", "No param", MessageBoxButtons.OK, MessageBoxIcon.Error) : Exit Sub
        If ForeValue.Text = "" Then MessageBox.Show("No Fore param", "No param", MessageBoxButtons.OK, MessageBoxIcon.Error) : Exit Sub

        ' - Select work directory -
        ProgressBar.Value = 0
        ScanFile.Text = ""
        Dim WorkDirName As String

        ' Select folder to work
        FolderBrowserDialog.Description = "Select where source html file's"
        If FolderBrowserDialog.ShowDialog() = Windows.Forms.DialogResult.Cancel Then
            Exit Sub
        End If
        WorkDirName = FolderBrowserDialog.SelectedPath

        If System.IO.Directory.Exists(WorkDirName + "\fixed_html") = False Then
            System.IO.Directory.CreateDirectory(WorkDirName + "\fixed_html")
        Else
            MessageBox.Show("Fixed folder already exist. Delete it or rename", "Exist output folder", MessageBoxButtons.OK, MessageBoxIcon.Error)
            Exit Sub
        End If

        ' Read files list
        Dim FileList() As String
        FileList = System.IO.Directory.GetFiles(FolderBrowserDialog.SelectedPath + "\", "*.htm", IO.SearchOption.TopDirectoryOnly)
        If FileList.Length = 0 Then
            MessageBox.Show("Source directory empty!", "Empty directory", MessageBoxButtons.OK, MessageBoxIcon.Error)
            Exit Sub
        End If

        Dim ReadStr As String

        Dim TempFileNum As Integer
        Dim InLink As String

        For TempFileNum = 0 To FileList.Length - 1

            ScanFile.Text = System.IO.Path.GetFileName(FileList(TempFileNum))
            Me.Refresh()


TryAgain:
            Dim InFixFile As New System.IO.StreamReader(FileList(TempFileNum), System.Text.Encoding.Default, True, 1)
            ReadStr = InFixFile.ReadToEnd
            Dim FixStatus As Boolean = False

            While InStr(ReadStr, "<a action=") <> 0

                Dim Temp1 As Integer, Temp2 As Integer
                Temp1 = InStr(ReadStr, "<a action=")
                Temp2 = InStr(Temp1, ReadStr, "</a>") + 4
                If Temp2 < Temp1 Then
                    InFixFile.Close()
                    Shell("notepad.exe " + FileList(TempFileNum), AppWinStyle.NormalFocus)
                    MessageBox.Show("Fix file and press OK to try again")
                    GoTo TryAgain
                End If

                InLink = Mid(ReadStr, Temp1, Temp2 - Temp1)
                If StartTag.Text <> "" And EndTag.Text <> "" Then
                    ReadStr = Replace(ReadStr, InLink, StartTag.Text + ChgHtmlButStl(InLink) + EndTag.Text)
                Else
                    ReadStr = Replace(ReadStr, InLink, ChgHtmlButStl(InLink))
                End If
                FixStatus = True
            End While
            InFixFile.Close()

            If FixStatus = True Then
                Dim oExecFile As System.IO.Stream = New System.IO.FileStream(WorkDirName + "\fixed_html\" + System.IO.Path.GetFileName(FileList(TempFileNum)), _
                    IO.FileMode.Create, IO.FileAccess.Write)
                Dim outExecFile As System.IO.StreamWriter = New System.IO.StreamWriter(oExecFile, System.Text.Encoding.Unicode)
                outExecFile.Write(ReadStr)
                outExecFile.Flush()
                outExecFile.Close()
            End If

            ProgressBar.Value = CInt((TempFileNum + 1) * 100 / FileList.Length)

        Next

        ProgressBar.Value = 0
        ScanFile.Text = ""
        MessageBox.Show("Html directory now +" & FileList.Length & " enchant.")

    End Sub

    Function ChgHtmlButStl(ByVal InLink As String) As String

        Dim InStrArr(1) As String
        Dim Temp1 As Integer, Temp2 As Integer

        ' 1. link 
        Temp1 = InStr(InLink, "<a action=") + 11
        Temp2 = InStr(Temp1, InLink, Chr(34))
        InStrArr(0) = Mid(InLink, Temp1, Temp2 - Temp1)

        ' 2. link text
        Temp1 = Temp2 + 2
        Temp2 = InStr(Temp1, InLink, "</a>")
        InStrArr(1) = Mid(InLink, Temp1, Temp2 - Temp1)

        '  <button action="link hello.html" value="Hello Chat" width=200 height=50 back="background.jpg" fore="foreground.jpg">
        ChgHtmlButStl = "<button action=" + Chr(34) + InStrArr(0) + Chr(34) + " value=" + Chr(34) + InStrArr(1) + Chr(34) + " width=" + WidthValue.Text + " height=" + HeightValue.Text + " back=" + Chr(34) + BackValue.Text + Chr(34) + " fore=" + Chr(34) + ForeValue.Text + Chr(34) + ">"

    End Function

    Private Sub AboutButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles AboutButton.Click
        MessageBox.Show("Tool for convert html to new C3 design. Idea by Virgi" _
        + vbNewLine + "Tool change standart button to button with user define style (size, background and etc). Example. This message in html file" _
        + vbNewLine _
        + vbNewLine + "<center><a action=" + Chr(34) + "bypass -h recipe?id=159" + Chr(34) + ">[Create]</a><br><br></center></body></html>" _
        + vbNewLine _
        + vbNewLine + "will be this after updating:" + vbNewLine _
        + vbNewLine + "<center><td><button action=" + Chr(34) + "bypass -h recipe?id=159" + Chr(34) + " value=" + Chr(34) + "[Create]" + Chr(34) + " width=111 height=222 back=" + Chr(34) + "L2UI_CH3.Btn1_normalOn" + Chr(34) + " fore=" + Chr(34) + "L2UI_CH3.Btn1_normal" + Chr(34) + "></td><br><br></center></body></html>" _
        + vbNewLine _
        + vbNewLine + "with using: width=111 height=222 back=L2UI_CH3.Btn1_normalOn fore=L2UI_CH3.Btn1_normal in tool dialog starttag=<td> endtag=</td>")

    End Sub
End Class

File: L2ScriptMaker/Modules/Others/HtmlButtonDesigner.cs
using L2ScriptMaker.Extensions.VbCompatibleHelper;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace L2ScriptMaker.Modules.Others
{
	public partial class HtmlButtonDesigner : Form
	{
		public HtmlButtonDesigner()
		{
			InitializeComponent();
		}

		private void QuitButton_Click(object sender, EventArgs e)
		{
			this.Close();
		}

		private void StartButton_Click(object sender, EventArgs e)
		{

			// check input params
			if (string.IsNullOrEmpty(WidthValue.Text))
			{
				MessageBox.Show("No Width param", "No param", MessageBoxButtons.OK, MessageBoxIcon.Error); return;
			}
			if (string.IsNullOrEmpty(HeightValue.Text))
			{
				MessageBox.Show("No Height param", "No param", MessageBoxButtons.OK, MessageBoxIcon.Error); return;
			}
			if (string.IsNullOrEmpty(BackValue.Text))
			{
				MessageBox.Show("No Back param", "No param", MessageBoxButtons.OK, MessageBoxIcon.Error); return;
			}
			if (string.IsNullOrEmpty(ForeValue.Text))
			{
				MessageBox.Show("No Fore param", "No param", MessageBoxButtons.OK, MessageBoxIcon.Error); return;
			}

			// - Select work directory -
			ProgressBar.Value = 0;
			ScanFile.Text = "";
			string WorkDirName;

			// Select folder to work
			FolderBrowserDialog.Description = "Select where source html file's";
			if (FolderBrowserDialog.ShowDialog() == DialogResult.Cancel)
				return;
			WorkDirName = FolderBrowserDialog.SelectedPath;

			if (System.IO.Directory.Exists(WorkDirName + @"\fixed_html") == false)
				System.IO.Directory.CreateDirectory(WorkDirName + @"\fixed_html");
			else
			{
				MessageBox.Show("Fixed folder already exist. Delete it or rename", "Exist output folder", MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}

			// Read files list
			string[] FileList;
			FileList = System.IO.Directory.GetFiles(FolderBrowserDialog.SelectedPath + @"\", "*.htm", System.IO.SearchOption.TopDirectoryOnly);
			if (FileList.Length == 0)
			{
				MessageBox.Show("Source directory empty!", "Empty directory", MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}

			string ReadStr;

			int TempFileNum;
			string InLink;
			var loopTo = FileList.Length - 1;
			for (TempFileNum = 0; TempFileNum <= loopTo; TempFileNum++)
			{
				ScanFile.Text = System.IO.Path.GetFileName(FileList[TempFileNum]);
				this.Refresh();


			TryAgain:
				;
				var InFixFile = new System.IO.StreamReader(FileList[TempFileNum], System.Text.Encoding.Default, true, 1);
				ReadStr = InFixFile.ReadToEnd();
				bool FixStatus = false;

				while (Strings.InStr(ReadStr, "<a action=") != 0)
				{
					int Temp1;
					int Temp2;
					Temp1 = Strings.InStr(ReadStr, "<a action=");
					Temp2 = Strings.InStr(Temp1, ReadStr, "</a>") + 4;
					if (Temp2 < Temp1)
					{
						InFixFile.Close();
						// Interaction.Shell("notepad.exe " + FileList[TempFileNum], AppWinStyle.NormalFocus);
						Process.Start("notepad.exe", FileList[TempFileNum]);
						MessageBox.Show("Fix file and press OK to try again");
						goto TryAgain;
					}

					InLink = Strings.Mid(ReadStr, Temp1, Temp2 - Temp1);
					if (!string.IsNullOrEmpty(StartTag.Text) & !string.IsNullOrEmpty(EndTag.Text))
						ReadStr = Strings.Replace(ReadStr, InLink, StartTag.Text + ChgHtmlButStl(InLink) + EndTag.Text);
					else
						ReadStr = Strings.Replace(ReadStr, InLink, ChgHtmlButStl(InLink));
					FixStatus = true;
				}
				InFixFile.Close();

				if (FixStatus == true)
				{
					System.IO.Stream oExecFile = new System.IO.FileStream(WorkDirName + @"\fixed_html\" + System.IO.Path.GetFileName(FileList[TempFileNum]), System.IO.FileMode.Create, System.IO.FileAccess.Write);
					var outExecFile = new System.IO.StreamWriter(oExecFile, System.Text.Encoding.Unicode);
					outExecFile.Write(ReadStr);
					outExecFile.Flush();
					outExecFile.Close();
				}

				ProgressBar.Value = Conversions.ToInteger((TempFileNum + 1) * 100 / (double)FileList.Length);
			}

			ProgressBar.Value = 0;
			ScanFile.Text = "";
			MessageBox.Show("Html directory now +" + Conversions.ToString(FileList.Length) + " enchant.");
		}

		private string ChgHtmlButStl(string InLink)
		{
			string ChgHtmlButStlRet = default(string);
			var InStrArr = new string[2];
			int Temp1;
			int Temp2;

			// 1. link 
			Temp1 = Strings.InStr(InLink, "<a action=") + 11;
			Temp2 = Strings.InStr(Temp1, InLink, Conversions.ToString((char)34));
			InStrArr[0] = Strings.Mid(InLink, Temp1, Temp2 - Temp1);

			// 2. link text
			Temp1 = Temp2 + 2;
			Temp2 = Strings.InStr(Temp1, InLink, "</a>");
			InStrArr[1] = Strings.Mid(InLink, Temp1, Temp2 - Temp1);

			// <button action="link hello.html" value="Hello Chat" width=200 height=50 back="background.jpg" fore="foreground.jpg">
			ChgHtmlButStlRet = "<button action=" + Conversions.ToString((char)34) + InStrArr[0] + Conversions.ToString((char)34) + " value=" + Conversions.ToString((char)34) + InStrArr[1] + Conversions.ToString((char)34) + " width=" + WidthValue.Text + " height=" + HeightValue.Text + " back=" + Conversions.ToString((char)34) + BackValue.Text + Conversions.ToString((char)34) + " fore=" + Conversions.ToString((char)34) + ForeValue.Text + Conversions.ToString((char)34) + ">";
			return ChgHtmlButStlRet;
		}

		private void AboutButton_Click(object sender, EventArgs e)
		{
			MessageBox.Show("Tool for convert html to new C3 design. Idea by Virgi"
		   + Constants.vbNewLine + "Tool change standart button to button with user define style (size, background and etc). Example. This message in html file"
		   + Constants.vbNewLine
		   + Constants.vbNewLine + "<center><a action=" + Conversions.ToString((char)34) + "bypass -h recipe?id=159" + Conversions.ToString((char)34) + ">[Create]</a><br><br></center></body></html>"
		   + Constants.vbNewLine
		   + Constants.vbNewLine + "will be this after updating:" + Constants.vbNewLine
		   + Constants.vbNewLine + "<center><td><button action=" + Conversions.ToString((char)34) + "bypass -h recipe?id=159" + Conversions.ToString((char)34) + " value=" + Conversions.ToString((char)34) + "[Create]" + Conversions.ToString((char)34) + " width=111 height=222 back=" + Conversions.ToString((char)34) + "L2UI_CH3.Btn1_normalOn" + Conversions.ToString((char)34) + " fore=" + Conversions.ToString((char)34) + "L2UI_CH3.Btn1_normal" + Conversions.ToString((char)34) + "></td><br><br></center></body></html>"
		   + Constants.vbNewLine
		   + Constants.vbNewLine + "with using: width=111 height=222 back=L2UI_CH3.Btn1_normalOn fore=L2UI_CH3.Btn1_normal in tool dialog starttag=<td> endtag=</td>");
		}
	}
}
